application/lib/templates.dart

```diff
@@ -133,38 +133,37 @@ class TemplateLibrary extends StatelessWidget {
           header: 'Door threshold',
           atoms: <AtomDescription>[],
           icon: Icon(Icons.sensor_door),
-        ),
+        ),*/
       ],
     );
   }
 }
 
 class Blueprint extends StatelessWidget {
   const Blueprint({
-    Key key,
-    @required this.header,
-    @required this.icon,
-    @required this.atoms,
-  }) : assert(atoms != null),
-       super(key: key);
+    Key? key,
+    required this.header,
+    required this.icon,
+    required this.atoms,
+  }) : super(key: key);
 
   final String header;
   final Widget icon;
   final List<AtomDescription> atoms;
 
   void _handleCreate(BuildContext context) {
     assert(this.atoms.isNotEmpty);
-    final List<Atom> atoms = this.atoms.map<Atom>((AtomDescription description) => description.create(AtomsDisposition.of(context))).toList();
-    Atom _lookupAtom(String identifier, { Atom ignore }) {
+    final List<Atom> atoms = this.atoms.map<Atom>((AtomDescription description) => description.create(AtomsDisposition.of(context)!)).toList();
+    Atom _lookupAtom(String? identifier, { Atom? ignore }) {
       final List<Atom> matches = atoms
-        .where((Atom atom) => atom != ignore && atom.identifier.matches(identifier))
+        .where((Atom atom) => atom != ignore && atom.identifier!.matches(identifier!))
```

Comment:
either identifier is nullable or it's not :-)

Reply:
fixed